src/variants/lambda/common.mligo

```diff
@@ -0,0 +1,67 @@
+// SPDX-FileCopyrightText: 2022 Tezos Commons
+// SPDX-License-Identifier: LicenseRef-MIT-TC
+
+#if !VARIANT_COMMON
+#define VARIANT_COMMON
+
+#include "../../error_codes.mligo"
+#include "../../helper/unpack.mligo"
+#include "types.mligo"
+#include "../../types.mligo"
+
+let handler_exists(handler_name, ce : string * contract_extra) : bool =
+  Big_map.mem handler_name ce.lambdas
+
+let handler_exists_and_active(handler_name, ce : string * contract_extra) : bool =
+  match Big_map.find_opt handler_name ce.lambdas with
+    | Some (v) -> v.is_active
+    | None -> (failwith proposal_handler_not_found: bool)
+
+let common_proposal_check (propose_params, ce : propose_params * contract_extra) : unit =
+  match unpack_proposal_metadata(propose_params.proposal_metadata) with
+    | Add_handler ahp -> if handler_exists(ahp.name, ce) then (failwith proposal_handler_exists : unit) else unit
+    | Remove_handler rhp -> if handler_exists_and_active(rhp, ce) then unit else (failwith proposal_handler_not_found : unit)
+    | Execute_handler ehp -> (match Big_map.find_opt ehp.handler_name ce.lambdas with
+          | Some hinfo ->
+              if hinfo.is_active
+                then let
+                  hc_input = (ehp.packed_argument, ce.handler_storage)
+                  in hinfo.handler_check hc_input
+                else (failwith proposal_handler_not_found: unit)
+          | None -> (failwith proposal_handler_not_found: unit))
+
+let add_handler(ce, ahp : contract_extra * add_handler_param) : contract_extra =
+  if handler_exists(ahp.name, ce)
+    then
+     (failwith proposal_handler_exists: contract_extra)
+    else let
+      handler_info = { code = ahp.code ; handler_check = ahp.handler_check; is_active = true }
+      in {ce with lambdas = Big_map.add ahp.name handler_info ce.lambdas }
+
+let disable_handler(ce, rhp : contract_extra * remove_handler_param) : contract_extra =
+  match Big_map.find_opt rhp ce.lambdas with
+    | Some hinfo -> {ce with lambdas = Big_map.update rhp (Some {hinfo with is_active = false}) ce.lambdas }
+    | None -> ce
```

Comment:
I need to mention again that the documentation is quite lacking.

Here for example one might think that this could be a mistake. It isn't, because we decided for this logic, but it would help to have a comment here for the future as well.

More importantly, there is no document for `lambdaDAO` in the `docs` folder, nor existing docs have been updated to mention it. The only existing documentation right now is the draft spec used to implement this, but the usual docs should be added to this repo as well.

Please update the comments and documentation for this PR.

Reply:
I have added the doc for lambdaDAO and added some comments.